packages/next/src/next-devtools/dev-overlay/shared.ts

```diff
@@ -39,41 +39,40 @@ type FastRefreshState =
   /** No refresh in progress. */
   | { type: 'idle' }
   /** The refresh process has been triggered, but the new code has not been executed yet. */
-  | { type: 'pending'; errors: SupportedErrorEvent[] }
+  | { type: 'pending'; errors: readonly SupportedErrorEvent[] }
 
 export interface OverlayState {
-  nextId: number
-  buildError: string | null
-  errors: SupportedErrorEvent[]
-  refreshState: FastRefreshState
-  versionInfo: VersionInfo
-  notFound: boolean
-  buildingIndicator: boolean
-  renderingIndicator: boolean
-  staticIndicator: boolean
-  showIndicator: boolean
-  disableDevIndicator: boolean
+  readonly nextId: number
```

Comment:
Manually annotating instead of `Readonly<T>` to avoid having TS inline the whole `type`. A deep readonly helper type is potentially too expensive whereas manual annotation is cheap.